.github/workflows/rust.yml

```diff
@@ -91,3 +91,20 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - uses: EmbarkStudios/cargo-deny-action@v1
+
+  doc:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          profile: minimal
+      - run: cargo doc --all-features
+  
+  cargo-semver-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: obi1kenobi/cargo-semver-checks-action@v1
```

Comment:
This should work with the [default semver-check assumptions](https://github.com/obi1kenobi/cargo-semver-checks-action#cargo-semver-checks-action), given this is a mono-crate repo that follows the tagging convention (which we should not forget to use), but would open up the crate to a wild west of changes on every PR as soon as one commit bumps the version number. That is, once the version number is bumped, there's no longer any CI signal on say, PRs that change an API method, because the "major version change" has already been enacted.

That's why I think it may be more useful to have `cargo-semver-checks-action` trigger on publish, such as in https://github.com/MystenLabs/mysten-infra/pull/145

But YMMV.